Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2945,24 +2945,8 @@ extension BrowserViewController: IntroViewControllerDelegate {
 }
 
 extension BrowserViewController: FxAContentViewControllerDelegate {
-    func contentViewControllerDidSignIn(_ viewController: FxAContentViewController, data: JSON) {
```

Comment:
This PR re-implements https://bugzilla.mozilla.org/show_bug.cgi?id=1335491 – the only difference is that the the `FxALoginFlags` struct is passed around instead of the `JSON` or `FxALoginResponse`.

Following a successful (even if unverified) login, the push registration dance happens. Once the account is verified, and https://bugzilla.mozilla.org/show_bug.cgi?id=1333780 is implemented, we should be able to start syncing immediately.